### PR TITLE
LPD-44640 Add related values when evaluating the DDMExpression

### PR DESCRIPTION
**Pull request by carolmariaabb**
https://liferay.atlassian.net/browse/LPD-44640

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 1]
```diff
@@ -537,6 +548,80 @@ else if (GetterUtil.getBoolean(results.get("invalidScript"))) {
...
+				StringBundler.concat(
+					relationshipObjectField.getName(), StringPool.UNDERLINE,
+					entry.getKey()),
+				entry.getValue());
+		}
+	}
+
+	private void _addValues(
+			ObjectDefinition objectDefinition2, String script, long userId,
+			Map<String, Object> variables)
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:591]: I would rename it to `ObjectDefinition objectDefinition`.

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 2]
```diff
@@ -447,6 +454,10 @@ else if (StringUtil.equals(
 						objectValidationRuleEngine.getKey(),
 						ObjectValidationRuleConstants.ENGINE_TYPE_DDM)) {
 
+				_addValues(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:457]: I think we can make this method more specific to know which values are being added.
What do you think about `addRelatedObjectEntryValues()` ? Would you happen to have a suggestion?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by marcelabc** [comment]
I will trigger the CI to anticipate the results.

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:object

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java (hunk 2 above)
Comment (carolmariaabb) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:457]: I agree! Done!

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:report:96510867

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**